pkg/expr/classic/reduce.go

```diff
@@ -7,6 +7,8 @@ import (
 	"github.com/grafana/grafana/pkg/expr/mathexp"
 )
 
+type classicReducer string
```

Comment:
Total nit: Let's drop the nilOrNaN function down below all the functions that have a classicReducer receiver type. Then they type def and the functions are close together.